Enabled per channel quantized static linear/conv

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37829 Some TODO fixes.
* #37626 Perf optimization for conv and gemm kernels.
* #37624 Added per channel separate test cases for fc and deconv tests.
* #37623 Changes to enable per channel support on dynamic linear.
* **#37622 Enabled per channel quantized static linear/conv**
* #37621 Added per channel kernels for depthwise conv.
* #37620 Changes to enable per channel requant.
* #37619 Enable per channel zero point.
* #37618 Interface changes to enable per channel quant.

Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

kimishpatel committed:
Enabled per channel quantized static linear/conv

Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized test on qlinear and qconv.
Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

dreiss commented on aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp:
```diff
@@ -280,6 +280,11 @@ at::Tensor PackedLinearWeightsQnnp::apply_dynamic_impl(at::Tensor input) {
         (uint8_t*)qnnp_w_data,
         nullptr);
     packB = w.get();
+    // Need to move the check here since we are releasing the weights.
+    TORCH_CHECK(
+        orig_weight.qscheme() == at::kPerTensorAffine,
+        "quantized::linear_dynamic (qnnpack) only supports "
+        "Per Tensor Quantization Scheme");
```
I don't get this.  Can the scheme change between when you call prepack and here?  Also, I think I understand "move" in the context of this diff, but I don't think someone looking at the new version of the code in the future will understand what's moving here.

kimishpatel replied:
This will be removed by a later PR in the stack. I just split the PRs for ease of review and ease of dissecting any future failures. I do understand though it is a cryptic comment.
Regarding qscheme changing. I don't think this is possible. Do you have a reason to believe that it can?

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enabled per channel quantized static linear/conv"


Summary:
Enable channelwise quantized tests on qlinear and qconv. Dynmaic linear to follow.

Test Plan:
pytest test/quantization/test_quantized.py
pytest test/quantization/test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21339046](https://our.internmc.facebook.com/intern/diff/D21339046)

[ghstack-poisoned]
